### PR TITLE
UF-6158

### PR DESCRIPTION
**Pull request by AndreSoderlund**
Added schema validation, changed the flyway script to the new database structure. Made some minor tweaks in the endpoint paths.  Version 2.0 -> 3.0

### CODE REVIEW

File: src/main/resources/db/migration/V1__initial_structure.sql
Comment (ronnberg) [inline src/main/resources/db/migration/V1__initial_structure.sql:1]: Flyway has never been deployed before?

Comment (AndreSoderlund) [reply src/main/resources/db/migration/V1__initial_structure.sql:1]: It has, but we are changing database and starting it from scratch.

Comment (ronnberg) [reply src/main/resources/db/migration/V1__initial_structure.sql:1]: 👍

File: src/main/resources/application-junit.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 config:
   datasource:
-    url: http://nowhere.com
+    url: jdbc:tc:mariadb:10.6.4:///
```
Comment (ronnberg) [inline src/main/resources/application-junit.yml:3]: Is 10.6.4 the version used in test/prod?

File: src/test/java/se/sundsvall/incident/integration/lifebuoy/LifeBuoyIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	@Test
+	void sendLifeBuoyTest() throws JsonProcessingException {
+		var incident = createIncidentEntity();
+		when(mockMapper.toLifeBuoyRequest(any())).thenReturn(LifeBuoyRequestWrapper.builder().build());
+		when(mockClient.sendLifebuoy(any())).thenReturn(new ResponseEntity<String>(HttpStatus.OK));
+
+		lifeBuoyIntegration.sendLifeBuoy(incident);
+
+		verify(mockMapper).toLifeBuoyRequest(incident);
+		verify(mockClient).sendLifebuoy(any());
```
Comment (ronnberg) [inline src/test/java/se/sundsvall/incident/integration/lifebuoy/LifeBuoyIntegrationTest.java:41]: Verify with same() insted, by making LifeBuoyRequestWrapper.builder().build() a local variable.

File: src/main/java/se/sundsvall/incident/integration/db/entity/IncidentEntity.java [hunk 3]
```diff
@@ -72,15 +73,15 @@ public class IncidentEntity {
 	private Status status;
 
 	@ManyToOne
-	@JoinColumn(name = "category_id")
+	@JoinColumn(name = "category_id", foreignKey = @ForeignKey(name = "fk_incident_category"))
 	private CategoryEntity category;
 
 	@Builder.Default
 	@OneToMany(cascade = {CascadeType.PERSIST, CascadeType.MERGE})
 	@JoinTable(
```
Comment (cheezi0747) [inline src/main/java/se/sundsvall/incident/integration/db/entity/IncidentEntity.java:81]: Replace this with. 

```
@JoinColumn(name = "incident_id", referencedColumnName = "id",
	foreignKey = @ForeignKey(name = "fk_incident_attachment_incident_id"))
```
a attachment cant live without a IncidentEntity, there is no reason to involve a joined table in this situation.

**Review by cheezi0747 — APPROVED** [review]

**Pull request merged**